scripts/lib/CIME/case_setup.py

```diff
@@ -160,16 +160,12 @@ def _case_setup_impl(case, caseroot, clean=False, test_mode=False, reset=False):
 
             case.flush()
             check_lockedfiles(case)
-            env_mach_pes = case.get_env("mach_pes")
-            pestot = env_mach_pes.get_total_tasks(models)
-            logger.debug("at update TOTALPES = {}".format(pestot))
-            case.set_value("TOTALPES", pestot)
-            thread_count = env_mach_pes.get_max_thread_count(models)
-            cost_pes = env_mach_pes.get_cost_pes(pestot, thread_count, machine=case.get_value("MACH"))
-            case.set_value("COST_PES", cost_pes)
 
             case.initialize_derived_attributes()
 
+            cost_per_node = 16 if case.get_value("MACH") == "yellowstone" else case.get_value("MAX_MPITASKS_PER_NODE")
+            case.set_value("COST_PES", (case.num_nodes - case.spare_nodes) * cost_per_node)
```

Comment:
Important change.

Reply:
Perhaps we should not attempt to automatically set COST_PES - since it often ends up incorrect.   Instead maybe we should consider adding a COST_PER_NODE variable to the machine definition in config_machines.xml?